src/main/java/bookstore/service/BookServiceImpl.java

```diff
@@ -36,4 +37,19 @@ public List<BookDto> findAll() {
                 .map(bookMapper::toDto)
                 .toList();
     }
+
+    @Override
+    public BookDto updateBookById(Long id, CreateBookRequestDto requestDto) {
+        bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Book wasn't updated, because book with id: "
+                + id + "doesn't exist"));
+        Book book = bookMapper.toModel(requestDto);
+        book.setId(id);
+        return bookMapper.toDto(bookRepository.save(book));
```

Comment:
bad approach

you should fetch the desired book by id, then update it, and save afterwards

you can leverage mapstruct to perform update:
```
void updateBook(CreateBookRequestDto  dto, @MappingTarget Book book);
```

the final version will be like:
```
Book book = repo.findById(id).orElseThrow(...);
mapper.updateBook(dto, book);
return mapper.toDto(repo.save(book));
```